INFO:__main__:============================ record when total_tx_num reaches 100=====================
ERROR:__main__:Errored: 1 retry happened in running tx: RelCustomerTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635679158.288186000,0 encountered previous write with future timestamp 1635679158.288365000,0 within uncertainty interval `t <= (local=1635679158.291396000,0, global=1635679158.788186000,0)`; observed timestamps: [{2 1635679158.289784000,0} {3 1635679158.288186000,0} {4 1635679158.291396000,0} {5 1635679158.299281000,0}]: "sql txn" meta={id=923b7b97 pri=0.06400743 epo=0 ts=1635679158.288186000,0 min=1635679158.288186000,0 seq=0} lock=false stat=PENDING rts=1635679158.288186000,0 wto=false gul=1635679158.788186000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 64, in run_tx
    each_tx_time = operation(m_conn)
  File "cockroachDB_driver.py", line 142, in <lambda>
    lambda l_conn: m_tx_ins.related_customer_transaction(l_conn, m_params))
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 551, in related_customer_transaction
    ''', (order[0], w_id, d_id, w_id)
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635679158.288186000,0 encountered previous write with future timestamp 1635679158.288365000,0 within uncertainty interval `t <= (local=1635679158.291396000,0, global=1635679158.788186000,0)`; observed timestamps: [{2 1635679158.289784000,0} {3 1635679158.288186000,0} {4 1635679158.291396000,0} {5 1635679158.299281000,0}]: "sql txn" meta={id=923b7b97 pri=0.06400743 epo=0 ts=1635679158.288186000,0 min=1635679158.288186000,0 seq=0} lock=false stat=PENDING rts=1635679158.288186000,0 wto=false gul=1635679158.788186000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval

 ]
ERROR:__main__:Errored: 2 retry happened in running tx: RelCustomerTxParams, ErrorMsg: 
[ restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635679163.176770000,0 encountered previous write with future timestamp 1635679163.176770000,1 within uncertainty interval `t <= (local=1635679163.178440000,0, global=1635679163.676770000,0)`; observed timestamps: [{2 1635679163.178440000,0} {3 1635679163.176770000,0} {4 1635679163.179267000,0}]: "sql txn" meta={id=721c8d67 pri=0.00049043 epo=0 ts=1635679163.176770000,0 min=1635679163.176770000,0 seq=0} lock=false stat=PENDING rts=1635679163.176770000,0 wto=false gul=1635679163.676770000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval
 ], Traceback: 
[ Traceback (most recent call last):
  File "cockroachDB_driver.py", line 64, in run_tx
    each_tx_time = operation(m_conn)
  File "cockroachDB_driver.py", line 142, in <lambda>
    lambda l_conn: m_tx_ins.related_customer_transaction(l_conn, m_params))
  File "/Users/nailixing/Documents/NUS_Modules/CS5424_Distributed_Database/projects/CS5424/txs/tx_workloadA.py", line 551, in related_customer_transaction
    ''', (order[0], w_id, d_id, w_id)
psycopg2.errors.SerializationFailure: restart transaction: TransactionRetryWithProtoRefreshError: ReadWithinUncertaintyIntervalError: read at time 1635679163.176770000,0 encountered previous write with future timestamp 1635679163.176770000,1 within uncertainty interval `t <= (local=1635679163.178440000,0, global=1635679163.676770000,0)`; observed timestamps: [{2 1635679163.178440000,0} {3 1635679163.176770000,0} {4 1635679163.179267000,0}]: "sql txn" meta={id=721c8d67 pri=0.00049043 epo=0 ts=1635679163.176770000,0 min=1635679163.176770000,0 seq=0} lock=false stat=PENDING rts=1635679163.176770000,0 wto=false gul=1635679163.676770000,0
HINT:  See: https://www.cockroachlabs.com/docs/v21.1/transaction-retry-error-reference.html#readwithinuncertaintyinterval

 ]
INFO:__main__:Running Tx RelCustomerTxParams successful at the 3 time retry
INFO:__main__:============================ record when total_tx_num reaches 200=====================
INFO:__main__:============================ record when total_tx_num reaches 300=====================
INFO:__main__:============================ record when total_tx_num reaches 400=====================
INFO:__main__:============================ record when total_tx_num reaches 500=====================
INFO:__main__:============================ record when total_tx_num reaches 600=====================
INFO:__main__:============================ record when total_tx_num reaches 700=====================
INFO:__main__:============================ record when total_tx_num reaches 800=====================
INFO:__main__:============================ record when total_tx_num reaches 900=====================
INFO:__main__:============================ using file /opt/project_files/xact_files_A/27.txt=====================
INFO:__main__:{'NewOrderTxParams': 866, 'PopItemTxParams': 1, 'PaymentTxParams': 12, 'StockLevelTxParams': 1, 'DeliveryTxParams': 17, 'RelCustomerTxParams': 2, 'TopBalanceTxParams': 1, 'OrderStatusTxParams': 1}
INFO:__main__:{'NewOrderTxParams': 866, 'PopItemTxParams': 1, 'PaymentTxParams': 12, 'StockLevelTxParams': 1, 'DeliveryTxParams': 17, 'RelCustomerTxParams': 2, 'TopBalanceTxParams': 1, 'OrderStatusTxParams': 1}
INFO:__main__:============================ total time used: 2014.4400408267975 second =====================
